src/download.rs

```diff
@@ -68,35 +71,38 @@ pub fn download_file(url: &str, file_path: &Path, redirect_path: &Path) -> Resul
       }
       Ok(bytes_read) => {
         file.write_all(&buffer[..bytes_read])?;
-        downloaded += bytes_read as u64;
+        just_downloaded += bytes_read as u64;
+        let downloaded = file_size + just_downloaded;
 
         let elapsed = start.elapsed().as_secs_f64();
         let speed = if elapsed > 0.0 {
-          downloaded as f64 / elapsed
+          just_downloaded as f64 / elapsed
         } else {
           0.0
         };
         measurements.push_back(speed);
-        if measurements.len() > 10 {
+        if measurements.len() > MEASUREMENT_SIZE {
           measurements.pop_front();
         }
         let avg_speed = measurements.iter().sum::<f64>() / measurements.len() as f64;
-        let eta = if avg_speed > 0.0 {
-          (total_size as f64 - downloaded as f64) / avg_speed
+        let eta = if avg_speed > 1.0 && measurements.len() > (MEASUREMENT_SIZE / 2) {
+          Eta::Seconds((total_size as f64 - downloaded as f64) / avg_speed)
         } else {
-          0.0
+          Eta::Unknown
         };
 
-        let progress = (downloaded as f64 / total_size as f64 * 100.0).round() as i64;
-        if progress > last_reported_progress {
+        let progress = downloaded as f64 / total_size as f64;
+        if last_reported_progress.is_none()
+          || last_reported_progress.is_some_and(|x| progress > x + 0.001)
```

Comment:
Consider:
```suggestion
        if progress > last_reported_progress.unwrap_or_default() + 0.001 {
```

This won't report when progress < 0.1% but I think it's fine.